### PR TITLE
Add Method to get n_params and AIC/BIC for GaussianHMM

### PR DESCRIPTION
**Pull request by richy1996**
#### Reference Issues/PRs
None

#### What does this implement/fix? Explain your changes.
- Adds the methods `_n_parameters`, `bic`, and `aic` to the `GaussianHMM` class.
- Essentially I copied the implementation from [sklearn implementation for gaussian mixture model](https://github.com/scikit-learn/scikit-learn/blob/baf0ea25d6dd034403370fea552b21a6776bef18/sklearn/mixture/_gaussian_mixture.py)

#### Any other comments?
I haven't fully implemented the methods for the other classes, e.g. GMMHMM etc. but it's in the works.

### CODE REVIEW

**Comment by blckmaxima** [comment]
Thank you! 
* Adding an example of aic/bic being used for model selection would be great.
* Incorporate these methods into the existing test suites.

File: lib/hmmlearn/hmm.py [hunk 1]
```diff
@@ -309,6 +309,50 @@ def _do_mstep(self, stats):
                     self._covars_ = ((covars_prior + c_n) /
                                      (cvweight + stats['post'][:, None, None]))
 
+    def _n_parameters(self):
```
[the remark this answers is not in the document]
Comment (richy1996) [reply lib/hmmlearn/hmm.py:312]: yeah it seems so, would you like me to incorporate _get_n_fit_scalars_per_param into aic/bic and scrap _n_parameters?

[the remark this answers is not in the document]
Comment (blckmaxima) [reply lib/hmmlearn/hmm.py:312]: I think that makes sense, and is inline with what I think @anntzer was thinking too.

**Comment by blckmaxima** [comment]
We've added `aic()` and `bic()` in a separate MR: https://github.com/hmmlearn/hmmlearn/pull/500

**Pull request closed without merging**